ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/Node.hs

```diff
@@ -1137,3 +1112,16 @@ mkPartialLedgerConfigShelley transitionConfig maxMajorProtVer shelleyTriggerHard
               maxMajorProtVer
         , shelleyTriggerHardFork = shelleyTriggerHardFork
         }
+
+class
+  ( ShelleyBasedEra (ShelleyBlockLedgerEra blk)
+  , blk ~ ShelleyBlock (BlockProtocol blk) (ShelleyBlockLedgerEra blk)
+  ) => IsShelleyBlock blk
+instance ShelleyBasedEra era => IsShelleyBlock (ShelleyBlock proto era)
+
+type family ShelleyBlockLedgerEra blk where
+  ShelleyBlockLedgerEra (ShelleyBlock proto era) = era
```

Comment:
I think this should live in the shelley subdir somewhere, perhaps in `Shelley/Block.hs`

Reply:
Could even be used to share code with
https://github.com/IntersectMBO/ouroboros-consensus/blob/af4b27ed334a9a08de1be3c78e8a76f3acbc37b9/ouroboros-consensus-cardano/src/unstable-cardano-tools/Cardano/Tools/DBAnalyser/Block/Cardano.hs#L203-L207

Reply:
I'd argue for leaving it here for [the same reasons](https://github.com/IntersectMBO/ouroboros-consensus/pull/934#discussion_r1508821949).

Reply:
For some reason, I missed your last comment, @amesgen. Could I use `ShelleyBlockEra` instead?

Reply:
Also consider replacing `IsShelleyBlock` with `IsShelleyBlock` in https://github.com/IntersectMBO/ouroboros-consensus/blob/af4b27ed334a9a08de1be3c78e8a76f3acbc37b9/ouroboros-consensus-cardano/src/unstable-cardano-tools/Cardano/Tools/DBAnalyser/Block/Cardano.hs#L203-L207

Reply:
I didn't realize the snippet you sent was in `DBAnalyser`. I could introduce a new module and reuse definitions from there :+1: